src/sentry/monitors/logic/mark_ok.py

```diff
@@ -33,17 +21,45 @@ def mark_ok(checkin: MonitorCheckIn, ts: datetime):
         "next_checkin": next_checkin,
         "next_checkin_latest": next_checkin_latest,
     }
-    if checkin.status == CheckInStatus.OK:
-        if monitor_env.monitor.status != ObjectStatus.DISABLED:
-            params["status"] = MonitorStatus.OK
-        # in the future this will auto-resolve associated issues
-        if monitor_env.status != MonitorStatus.OK:
-            params["last_state_change"] = ts
-            # resolve any associated incidents
-            incidents = MonitorIncident.objects.filter(
-                monitor_environment=monitor_env, grouphash=monitor_env.incident_grouphash
+
+    # Don't update incidents + certain params if monitor is disabled or already in an OK state
+    if (
+        monitor_env.monitor.status != ObjectStatus.DISABLED
+        and monitor_env.status != MonitorStatus.OK
+    ):
+        params["status"] = MonitorStatus.OK
+        recovery_threshold = monitor_env.monitor.config.get("recovery_threshold", 0)
+
+        # Run incident logic if recovery threshold is set
+        if recovery_threshold:
+            # Safety check to prevent db lockup on check-in lookup
+            assert recovery_threshold is not None
```

Comment:
this will raise an exception into sentry right?

Reply:
Yeah but it will never ever happen because of the if statement